inox2d/src/node/components.rs

```diff
@@ -0,0 +1,31 @@
+/*!
+Inochi2D node types to Inox2D components:
+- Node -> (Nothing)
+- Part -> Drawable + TexturedMesh
+- Composite -> Drawable + Composite
+- SimplePhysics -> SimplePhysics
+- Custom nodes by inheritance -> Custom nodes by composition
+*/
+
+pub mod composite;
+pub mod drawable;
+pub mod simple_physics;
+pub mod textured_mesh;
+
+/// Internal component solving for deforms of a node.
+pub(crate) mod deform_stack;
+/// Internal component storing:
+/// - Relative transform being determined in between frames.
+/// - Absolute transform prepared from all relative transforms just before rendering.
+pub(crate) mod transform_store;
+/// Internal component storing zsort being determined in between frames.
+pub(crate) mod zsort;
+
+pub use composite::Composite;
+pub use drawable::Drawable;
+pub use simple_physics::SimplePhysics;
+pub use textured_mesh::TexturedMesh;
+
+pub(crate) use deform_stack::DeformStack;
+pub(crate) use transform_store::TransformStore;
+pub(crate) use zsort::ZSort;
```

Comment:
Given the respective sizes of these modules, I think all of the structures defined in there should live in this single file instead. Right now it feels artificially sparse, the individual structures don't have any complex logic on their own that would be contained in separate files, making the modules shallow.

Reply:
This is a valid point, as component definition files indeed should only contain data and no logic.
And I should probably put `DeformStack` logic to somewhere else, likely under `render`, cause this is indeed *some* logic.